File: ROS Messages And Topics/ros_topic_and_messages_pkg/script/subscribe_robot.py
#!/usr/bin/env python3

import rospy
from ros_topic_and_messages_pkg.msg import Position

def callback(data):
    rospy.loginfo("Received: x=%f, y=%f, theta=%f", data.x, data.y, data.theta)

def subscriber():
    rospy.init_node('position_subscriber', anonymous=True)
    rospy.Subscriber('robot_position', Position, callback)
    rospy.spin()

if __name__ == '__main__':
    subscriber()

File: ROS-Messages-and-Topics/ros_topic_and_messages_pkg/src/subscribe_robot.cpp
#include <ros/ros.h>
#include <ros_topic_and_messages_pkg/Position.h>  

void callback(const ros_topic_and_messages_pkg::Position::ConstPtr& data)
{
    ROS_INFO("Received: x=%f, y=%f, theta=%f", data->x, data->y, data->theta);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "position_subscriber");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("robot_position", 10, callback);
    ros::spin();

    return 0;
}